### DISCUSSION TITLE
Customizing key removes which-key descriptions for other keys

### DISCUSSION THREAD

**Discussion by weberam2** [LazyVim/LazyVim#4564] [General]
Hello,

I'm trying to configure a key to only work with the R lang LazyExtra package

I have an R.lua file in plugins:

```
return {
  "R-nvim/R.nvim",
  opts = {
    rmdchunk = 0, --- prevent issues with autocomplete when you type `
  },
  keys = {
    {
      "<localleader>z",
      mode = "v",
      ":! tr -s \" \" | column -t -s '\"' -o '\"' <CR>",
      desc = "Adjust R tables",
    },
  },
}
```

This works, but suddenly the other descriptions disappear, and it isn't SPECIFIC to R.nvim

![image](https://github.com/user-attachments/assets/bfff7778-2318-4365-a7bb-5604fca26a2e)

Perhaps there's another way where it is only activate for a specific file type? .rmd or .r or something like that?

Thank you for your help

**Comment by dpetka2001** [comment]
The descriptions are just not available in visual mode from the initial configuration. I created #4565 that will fix that. As for your second question, you can use `ft = "r"` in your keymap to make it available only for a specific filetype.

**Reply by weberam2** [reply]
Oh! I feel stupid for not noticing that before. Wild!
Thanks for fixing it!

So is what you're suggesting like this?

```
return {
  "R-nvim/R.nvim",
  opts = {
    rmdchunk = 0, --- prevent issues with autocomplete when you type `
  },
  keys = {
    {
      "<localleader>z",
      ft = "r",
      mode = "v",
      ":! tr -s \" \" | column -t -s '\"' -o '\"' <CR>",
      desc = "Adjust R tables",
    },
  },
}
```

or 

keymaps.lua:

```
local map = vim.keymap.set
map("<localleader>z", ft = "r", mode = "v", "<Esc>la",  ":! tr -s \" \" | column -t -s '\"' -o '\"' <CR>", { desc = "Escape brackets" })
```
or something like that...

**Reply by dpetka2001** [reply]
No in `keymaps.lua` i believe won't work for specifying a filetype. There's no such field in `vim.keymap.set` if i'm not mistaken. You would either have to do it in an autocmd on `FileType` event or just put the `vim.keymap.set` in a file in `~/.config/nvim/after/ftplugin/r.lua`. Or just put the `ft = "r"` in your `keys` section.
